apps/server/src/modules/account/services/account-db.service.ts

```diff
@@ -1,110 +1,113 @@
 		const accountEntity = await this.accountRepo.findByUserIdOrFail(userId);
-		return AccountEntityToDtoMapper.mapToDto(accountEntity);
+		return AccountEntityToDoMapper.mapToDto(accountEntity);
 	}
 
-	async findByUsernameAndSystemId(username: string, systemId: EntityId | ObjectId): Promise<AccountDto | null> {
+	async findByUsernameAndSystemId(username: string, systemId: EntityId | ObjectId): Promise<Account | null> {
 		const accountEntity = await this.accountRepo.findByUsernameAndSystemId(username, systemId);
-		return accountEntity ? AccountEntityToDtoMapper.mapToDto(accountEntity) : null;
+		return accountEntity ? AccountEntityToDoMapper.mapToDto(accountEntity) : null;
 	}
 
-	async save(accountDto: AccountSaveDto): Promise<AccountDto> {
-		let account: Account;
+	async save(accountSave: AccountSave): Promise<Account> {
+		let account: AccountEntity;
 		// HINT: mapping could be done by a mapper (though this whole file is subject to be removed in the future)
 		// HINT: today we have logic to map back into unit work in the baseDO
-		if (accountDto.id) {
-			const internalId = await this.getInternalId(accountDto.id);
+		if (accountSave.id) {
+			const internalId = await this.getInternalId(accountSave.id);
 			account = await this.accountRepo.findById(internalId);
-			account.userId = new ObjectId(accountDto.userId);
-			account.systemId = accountDto.systemId ? new ObjectId(accountDto.systemId) : undefined;
-			account.username = accountDto.username;
-			account.activated = accountDto.activated;
-			account.expiresAt = accountDto.expiresAt;
-			account.lasttriedFailedLogin = accountDto.lasttriedFailedLogin;
-			if (accountDto.password) {
-				account.password = await this.encryptPassword(accountDto.password);
+			account.userId = new ObjectId(accountSave.userId);
+			account.systemId = accountSave.systemId ? new ObjectId(accountSave.systemId) : undefined;
+			account.username = accountSave.username;
+			account.activated = accountSave.activated;
+			account.expiresAt = accountSave.expiresAt;
+			account.lasttriedFailedLogin = accountSave.lasttriedFailedLogin;
+			if (accountSave.password) {
+				account.password = await this.encryptPassword(accountSave.password);
 			}
-			account.credentialHash = accountDto.credentialHash;
-			account.token = accountDto.token;
+			account.credentialHash = accountSave.credentialHash;
+			account.token = accountSave.token;
 
 			await this.accountRepo.save(account);
 		} else {
-			account = new Account({
-				userId: new ObjectId(accountDto.userId),
-				systemId: accountDto.systemId ? new ObjectId(accountDto.systemId) : undefined,
-				username: accountDto.username,
-				activated: accountDto.activated,
-				expiresAt: accountDto.expiresAt,
-				lasttriedFailedLogin: accountDto.lasttriedFailedLogin,
-				password: accountDto.password ? await this.encryptPassword(accountDto.password) : undefined,
-				token: accountDto.token,
-				credentialHash: accountDto.credentialHash,
+			account = new AccountEntity({
```

Comment:
Can we switch this to the Account DO?

Reply:
Account DO requires an id which does not exist at this point

Reply:
use either Omit<Account> to circumvent this or create a new id with new ObjectId().toHexString(). this is already used in code